core/sql/common/CharType.cpp

```diff
@@ -752,16 +752,21 @@ NABoolean CharType::isCharSetAndCollationComboOK() const
 // -----------------------------------------------------------------------
 // Equality comparison
 // -----------------------------------------------------------------------
+NABoolean CharType::compareCharAttrs(const NAType& other) const
+{
```

Comment:
By the time this method is called, it has already gone through the test of
validating that 'this' and 'other' are part of same class.